crates/ruma-macros/src/lib.rs

```diff
@@ -108,27 +215,195 @@ pub fn event_enum(input: TokenStream) -> TokenStream {
+///   traits, even if it is only a type alias, and the `RedactContent` trait must still be
+///   implemented for those kinds.
+/// * `PossiblyRedactedFooEventContent`: the form of the event content that is used when we don't
+///   know whether a `State` event is redacted or not. It means that on this type any field that is
+///   redacted must be optional, or it must have the `#[serde(default)]` attribute for
+///   deserialization.
 ///
-/// Also generates type aliases depending on the kind of event, with the final `Content` of the type
-/// name removed and prefixed added. For instance, a message-like event content type
-/// `FooEventContent` will have the following aliases generated:
+///   The generated type implements `PossiblyRedactedStateEventContent`, `StaticEventContent`,
+///   `Serialize` and `Deserialize`.
 ///
-/// * `type FooEvent = MessageLikeEvent<FooEventContent>`
-/// * `type SyncFooEvent = SyncMessageLikeEvent<FooEventContent>`
-/// * `type OriginalFooEvent = OriginalMessageLikeEvent<FooEventContent>`
-/// * `type OriginalSyncFooEvent = OriginalSyncMessageLikeEvent<FooEventContent>`
-/// * `type RedactedFooEvent = RedactedMessageLikeEvent<FooEventContent>`
-/// * `type RedactedSyncFooEvent = RedactedSyncMessageLikeEvent<FooEventContent>`
+///   The generation uses the rules as the redacted type, using the `#[ruma_event(skip_redaction)]`
+///   attribute.
+///
+///   To skip the generation of this type to use a custom type, the
+///   `#[ruma_event(custom_possibly_redacted)]` attribute can be used on the container. The
+///   `PossiblyRedactedFooEventContent` type must still exist for the `State` kind and implement the
+///   same traits, even if it is only a type alias.
+///
+/// Event content types of the `MessageLike` kind that use the `Relation` type also need a clone of
+/// the event content without the `relates_to` field for use within relations, where nested
+/// relations are not meant to be serialized by homeservers. This macro can generate a
+/// `FooEventContentWithoutRelation` type if the `#[ruma_event(without_relation)]` attribute is
+/// applied on the container. It also generates `From<FooEventContent> for
+/// FooEventContentWithoutRelation` and `FooEventContentWithoutRelation::with_relation()`.
+///
+/// By default, the generated types get a `#[non_exhaustive]` attribute. This behavior can be
+/// controlled by setting the `ruma_unstable_exhaustive_types` compile-time `cfg` setting as
+/// `--cfg=ruma_unstable_exhaustive_types` using `RUSTFLAGS` or `.cargo/config.toml` (under
+/// `[build]` -> `rustflags = ["..."]`). When that setting is activated, the attribute is not
+/// applied so the types are exhaustive.
+///
+/// # Type aliases
+///
+/// All kinds generate at least one type alias for the full event format. For the same example type
+/// named `FooEventContent`, the type alias generated is `type FooEvent =
```

Comment:
"the type alias" sounds like there is only one. How about

```suggestion
/// named `FooEventContent`, the first type alias generated is `type FooEvent =
```

Reply:
Applied.